OneCommunityGlobal/HighestGoodNetworkApp
# Ilya - Tools table

<img width="771" alt="tool-list-task" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105448137/3ba8c506-25a9-45ee-a12d-e623120b60af">

# Description
Adds functionality for the user to navigate to [/bmdashboard/tools](http://localhost:3000/bmdashboard/tools), and view a list of all tool items currently in the system. The table can be sorted by tool name or by project and the results can be filtered by tool name. Update History and Purchase History can be viewed for each item in a modal. 

## Related PRS (if any):
This frontend PR is related to the [#913](https://github.com/OneCommunityGlobal/HGNRest/pull/913) backend PR.
To test this frontend PR you need to checkout the #913 backend PR.
…

## Main changes explained:
- A new route was added to routes.js to correspond to "/bmdashboard/tools"
- ToolsList.jsx, ToolItemListView.jsx, ToolItemsTable.jsx, ToolRecordsModal.jsx and UpdateToolModal.jsx components were created
- New redux action created to make an API call and receive a list of tool items
- Table columns correspond to the information found on tool item schema, which is vastly different from Material or Consumable items

## How to test:
1. Check out **ilya_tools_list branch** on the back end. Make sure you are using Node version 14.17.6. Run **npm install** and **npm run build && npm start** to run the backend locally
2. Check out the current branch on the frontend. Run commands **npm install** and **npm run start:local** to launch the front end application
3. Log in as any user
4. Go to [/bmdashboard](http://localhost:3000/bmdashboard) and log in again
5. Navigate to [/bmdashboard/tools](http://localhost:3000/bmdashboard/tools). A list of all tools currently in the system should display. 
6. Test the filtering by Project and Tool function (the dropdowns above the table)
7. Test sorting by project and by name (the arrows next to "Project" and "Name" in the table header
8. Pressing the View button under Updates should open an Updates Record modal (not all items have information to display)
9. Pressing the View button under Purchases should open an Purchase Record modal (not all items have information to display)
10. Pressing the pencil should open an Update Tool Form modal. The content of this modal is outside of the scope of this task

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105448137/fdd8c961-0d34-4219-b1c5-743aef1c2820
